client/src/components/my-mentor-card/index.tsx

```diff
@@ -52,16 +56,13 @@ const useStyles = makeStyles(() => ({
 export default function MyMentorCard(props: {
   editDisabled: boolean;
   continueAction: () => void;
-  useMentor: UseWithMentor;
+  useMentor: UseMentorEdits;
 }): JSX.Element {
-  const {
-    mentor,
-    isMentorLoading,
-    isMentorSaving,
-    mentorError,
-    editedMentor,
-    editMentor,
-  } = props.useMentor;
+  const mentor = useActiveMentor((state) => state.data);
```

Comment:
you don't need the whole mentor here. In fact, you already have a function that you should use as a selector...


```typescript
  /**
  const mentor = useActiveMentor((state) => state.data);
  const mentorError = useActiveMentor((state) => state.error);
  // generally defer calling any of these isActiveMentorXXX
  // functions until the place you actually need the result
  const isMentorLoading = isActiveMentorLoading();
  const isMentorSaving = isActiveMentorSaving();
  const { editedMentor, editMentor } = props.useMentor;

  if (!mentor || !editedMentor) {
    return <div />;
  }
  const mentorInfo = parseMentor(mentor);
  */
  const mentorId = useActiveMentor(ms => ms.data?._id || "");
  if (!(mentorId && editedMentor)) {
    return <div />;
  }
  const mentorInfo = useActiveMentor(ms => parseMentor(ms.data));
```